osu.Game/Online/Rooms/RoomCategory.cs

```diff
@@ -13,5 +13,8 @@ public enum RoomCategory
 
         [Description("Featured Artist")]
         FeaturedArtist,
+
+        [Description("Daily Challenge")]
+        DailyChallenge,
```

Comment:
As mentioned in https://github.com/ppy/osu-web/pull/11204, this is required to make `daily_challenge` rooms deserialise at all. Without this change deserialisation just dies, which is a bit unfortunate if/once old clients ever come in contact with these - but now it is too late to do something about it anyway.